### PR TITLE
Put back C++ JWTCredentials code and add a test case in interop test.

### PR DESCRIPTION
**Pull request by yang-g**
Somehow the code was dropped somewhere along the road...

### CODE REVIEW

**Comment by jboeuf** [comment]
LGTM but looks like python is broken. I'm sure why though. Maybe @nathanielmanistaatgoogle could have a look: https://travis-ci.org/grpc/grpc/jobs/54016096

File: src/cpp/client/secure_credentials.cc [hunk 1]
```diff
@@ -104,6 +104,15 @@ std::unique_ptr<Credentials> ServiceAccountCredentials(
       json_key.c_str(), scope.c_str(), lifetime));
 }
 
+// Builds JWT credentials.
+std::unique_ptr<Credentials> JWTCredentials(
+    const grpc::string &json_key, std::chrono::seconds token_lifetime) {
+  gpr_timespec lifetime = gpr_time_from_seconds(
+      token_lifetime.count() > 0 ? token_lifetime.count() : 0);
```
Comment (jboeuf) [inline src/cpp/client/secure_credentials.cc]: Does it even make sense to have a lifetime == 0. Shouldn't we just fail there?

Comment (yang-g) [inline src/cpp/client/secure_credentials.cc]: So I guess the problem is which layer should fail. This code is expecting the c code returning a NULL. Do you think it is better to error out here?

**Comment by yang-g** [comment]
The python build seems fine to me?

File: src/cpp/client/secure_credentials.cc (hunk 1 above)
Comment (jboeuf) [inline src/cpp/client/secure_credentials.cc]: I think that it's better to fail early. It's easier to debug. What do you think?

**Comment by jboeuf** [comment]
Python build is now fine. It was broken when I made the comment.

File: src/cpp/client/secure_credentials.cc (hunk 1 above)
Comment (yang-g) [inline src/cpp/client/secure_credentials.cc]: Added a check (and also for the service account one)

File: src/cpp/client/secure_credentials.cc [hunk 2]
```diff
@@ -98,12 +98,25 @@ std::unique_ptr<Credentials> ComputeEngineCredentials() {
 std::unique_ptr<Credentials> ServiceAccountCredentials(
     const grpc::string& json_key, const grpc::string& scope,
     std::chrono::seconds token_lifetime) {
-  gpr_timespec lifetime = gpr_time_from_seconds(
-      token_lifetime.count() > 0 ? token_lifetime.count() : 0);
+  if (token_lifetime.count() <= 0) {
+    return WrapCredentials(nullptr);
```
Comment (jboeuf) [inline src/cpp/client/secure_credentials.cc]: One more thing... Should we log in this case?

**Comment by yang-g** [comment]
Done.

**Pull request merged**